src/computation/logic/distribution.ts

```diff
@@ -0,0 +1,103 @@
+import { copyDictionary, Dictionary } from "../../utilities/dictionary";
+import { SortedReverseDict, KeyValuePair } from "./sorted-reverse-dict";
+
+/**
+ * A general function for distributing a number of items on a number of names based on updated quotients.
+ *
+ * Each time a name wins an item, their count of items won is updated.
+ * Subsequently the denominator used to calculate their quotient is updated to reflect their win on whether or not they will win following items.
+ *
+ * @param numberToDistribute Number of items to distribute
+ * @param distributeOn A dictionary of names to distribute on, if there is already a partial distribution the values in this dictionary will be used as a starting point for how many items they have already received
+ * @param baseValue The value used as the numerator for each name
+ * @param denominatorFunction The function used to select the denominator to be used depending on how many items each name has won
+ */
+export function distributionByQuotient(
+    numberToDistribute: number,
+    distributeOn: Dictionary<number>,
+    baseValue: Dictionary<number>,
+    denominatorFunction: (timesWon: number) => number
+): Dictionary<number> {
+    const updatedDistribution = copyDictionary(distributeOn);
+    const sortedQuotients = new SortedReverseDict();
+
+    // Fill the sortedQuotients with all the entries expected to have the starting list fully sorted
+    for (const entry in updatedDistribution) {
+        if (updatedDistribution.hasOwnProperty(entry)) {
+            const quotient = baseValue[entry] / denominatorFunction(updatedDistribution[entry]);
+            sortedQuotients.insert({ key: entry, value: quotient });
+        }
+    }
+
+    // Begin drawing winners
```

Comment:
Should probably be a function (something one should think about every time one writes a descriptive title in a comment rather than pointing out something unusual or important).

Reply:
Simplified the function by abstracting the functionality into multiple functions.